crates/ra_ide_db/src/defs.rs

```diff
@@ -47,7 +47,10 @@ impl Definition {
         match self {
             Definition::Macro(_) => None,
             Definition::Field(sf) => Some(sf.visibility(db)),
-            Definition::ModuleDef(def) => module?.visibility_of(db, def),
+            Definition::ModuleDef(def) => match def {
+                ModuleDef::EnumVariant(id) => Some(id.visibility(db)),
```

Comment:
On the other other hand, you would already get that by just looking up the visibility of the containing enum here, no need to implement `HasVisibility` or change `EnumData`.

Reply:
right, thanks, this makes more sense